src/tools/rbd/action/Export.cc

```diff
@@ -86,12 +365,156 @@ class C_Export : public Context
   SimpleThrottle &m_throttle;
   librbd::Image &m_image;
   bufferlist m_bufferlist;
+  uint64_t m_dest_offset;
   uint64_t m_offset;
   uint64_t m_length;
   int m_fd;
 };
 
-static int do_export(librbd::Image& image, const char *path, bool no_progress)
+static int do_export_v2(librbd::Image& image, librbd::image_info_t &info, int fd,
+		        uint64_t period, int max_concurrent_ops, utils::ProgressContext &pc)
+{
+  int r = 0;
+  // header
+  bufferlist bl;
+  bl.append(utils::RBD_IMAGE_BANNER_V2);
+
+  // size in header
+  uint64_t size = info.size;
+  ::encode(size, bl);
```

Comment:
This is needed for V1 import. When we do a v1 import, there is no diffs, and we need to know the size of the whole image.